test/parallel/test-async-wrap-asyncresource-constructor.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+
+// This tests that AsyncResource throws an error if bad parameters are passed
+
+const assert = require('assert');
+const AsyncResource = require('async_hooks').AsyncResource;
+const async_wrap = process.binding('async_wrap');
+
+assert.throws(() => {
+  return new AsyncResource();
+}, new RegExp('^TypeError: type must be a string with length > 0$'));
```

Comment:
No, my comments were not addressed. I would prefer to see the `/..../` regex pattern used but I'm fine with this either way.